cpp/mrc/src/public/segment/builder.cpp

```diff
@@ -41,6 +42,23 @@ std::string accum_merge(std::string lhs, std::string rhs)
 }  // namespace
 
 namespace mrc::segment {
+
+void Builder::init_module(std::shared_ptr<mrc::modules::SegmentModule> smodule)
+{
+    ns_push(smodule);
+    VLOG(2) << "Initializing module: " << m_namespace_prefix;
+    smodule->m_module_instance_registered_namespace = m_namespace_prefix;
+    smodule->initialize(*this);
+    ns_pop();
+
+    // TODO(Devin): Maybe a better way to do this with compile time type ledger.
+    if (std::dynamic_pointer_cast<modules::PersistentModule>(smodule) != nullptr)
```

Comment:
What is a `PersistentModule`? How is it different?

Reply:
Its persistent/stateful. By default modules are stateless and can be released once they're initialized. A PersistentModule might have common member data; one motivating example is the stream buffer module that holds a boost circular buffer.